lib/modules/hosts/derpibooru.js

```diff
@@ -0,0 +1,38 @@
+addLibrary('mediaHosts', 'derpibooru', {
+	name: 'Derpibooru',
+	domains: ['derpiboo.ru'],
+	matchRe: /^https?:\/\/(?:www\.)?derpiboo.ru\/\d+$/i,
+
+	detect: function(href, _elem) {
+		return modules['showImages'].siteModules['derpibooru'].matchRe.test(href);
+	},
+
+	handleLink: function(elem) {
+		var def = $.Deferred();
+		var apiURL = elem.href + '.json';
+		RESEnvironment.ajax({
+			method: 'GET',
+			url: apiURL,
+			onload: function(response) {
+				var json;
+				try {
+					json = JSON.parse(response.responseText);
+				} catch (error) {
+					def.reject();
+				}
+				def.resolve(elem, json);
+			},
+			onerror: function(response) {
+				def.reject();
+			}
+		});
+		return def.promise();
+	},
+
+	handleInfo: function(elem, info) {
+		elem.type = 'IMAGE';
+		elem.src = info.image;
```

Comment:
Ah, thanks. I was having trouble finding a place with a lot of Derpibooru images to test on. (/r/mylittlepony usually only has a lot of Derpibooru images immediately after a new episode comes out). I'll definitely give the `duplicate_of` bit a shot.